gr-digital/lib/crc32_bb_impl.cc

```diff
@@ -65,21 +65,40 @@ namespace gr {
       }
     }
 
+    unsigned int
+    crc32_bb_impl::calculate_crc32(const unsigned char* in, size_t packet_length){
+      unsigned int crc = 0;
+      d_crc_impl.reset();
+      if (!d_packed){
+        const size_t n_packed_length = 1 + ((packet_length - 1) / 8);
```

Comment:
Are we 100.000001% sure that `packet_length != 0` at this point?

Reply:
Hm, we could test that and throw? or use some less efficient algorithm? From below: if packet_length==crc_length && d_check this might actually do harm, but is this realistic?